pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -304,4 +310,153 @@ var _ = Describe("TaskRun Reconciler General Tests", func() {
+			metricDto := &dto.Metric{}
+			var pmetrics *mpcmetrics.PlatformMetrics
+			mpcmetrics.HandleMetrics(platform, func(metrics *mpcmetrics.PlatformMetrics) {
+				pmetrics = metrics
+			})
+			Expect(pmetrics).ShouldNot(BeNil())
+			Expect(pmetrics.CleanupFailures.Write(metricDto)).ShouldNot(HaveOccurred())
+			return metricDto.GetCounter().GetValue()
+		}
+
+		It("should increment CleanupFailures metric when cleanup task fails", func(ctx SpecContext) {
+			// Get initial metric value
+			initialFailures := getCleanupFailuresMetric(platform)
+
+			// Create and assign a TaskRun to a host
+			createUserTaskRun(ctx, client, "test-cleanup-failure", platform)
+
+			// First reconciliation - should assign host
+			_, err := reconciler.Reconcile(ctx, reconcile.Request{
+				NamespacedName: types.NamespacedName{Namespace: userNamespace, Name: "test-cleanup-failure"},
+			})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Verify TaskRun was assigned to a host
+			tr := getUserTaskRun(ctx, client, "test-cleanup-failure")
+			Expect(tr.Labels[AssignedHost]).ShouldNot(BeEmpty())
+
+			// Complete the TaskRun (mark it as successful)
+			tr.Status.CompletionTime = &metav1.Time{Time: time.Now()}
+			tr.Status.SetCondition(&apis.Condition{
+				Type:   apis.ConditionSucceeded,
+				Status: corev1.ConditionTrue,
+				Reason: "Succeeded",
+			})
+			Expect(client.Status().Update(ctx, tr)).Should(Succeed())
+
+			// Second reconciliation - should trigger cleanup
+			_, err = reconciler.Reconcile(ctx, reconcile.Request{
+				NamespacedName: types.NamespacedName{Namespace: userNamespace, Name: "test-cleanup-failure"},
+			})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Find the cleanup task that should have been created
+			cleanupTasks := &pipelinev1.TaskRunList{}
+			err = client.List(ctx, cleanupTasks, runtimeclient.MatchingLabels{
+				TaskTypeLabel:     TaskTypeClean,
+				UserTaskName:      "test-cleanup-failure",
+				UserTaskNamespace: userNamespace,
+			})
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(len(cleanupTasks.Items)).Should(Equal(1))
+
+			cleanupTask := &cleanupTasks.Items[0]
+
+			// Mark the cleanup task as failed
+			cleanupTask.Status.CompletionTime = &metav1.Time{Time: time.Now()}
+			cleanupTask.Status.SetCondition(&apis.Condition{
+				Type:   apis.ConditionSucceeded,
+				Status: corev1.ConditionFalse,
+				Reason: "CleanupFailed",
+			})
+			Expect(client.Status().Update(ctx, cleanupTask)).Should(Succeed())
+
+			// Reconcile the cleanup task - this should increment the CleanupFailures metric
+			_, err = reconciler.Reconcile(ctx, reconcile.Request{
+				NamespacedName: types.NamespacedName{Namespace: cleanupTask.Namespace, Name: cleanupTask.Name},
+			})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Verify the CleanupFailures metric incremented
+			newFailures := getCleanupFailuresMetric(platform)
+			Expect(newFailures).Should(Equal(initialFailures + 1))
```

Comment:
In this spirit of @sadlerap  - this can be a one-liner
```suggestion
			Expect(getCleanupFailuresMetric(platform)).Should(Equal(initialFailures + 1))
```